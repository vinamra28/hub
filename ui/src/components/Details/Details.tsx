/*import React from 'react';
import {
  Card,
  CardHeader,
  Grid,
  GridItem,
  Text,
  TextContent,
  TextVariants,
  Badge
} from '@patternfly/react-core';
import { BuildIcon, UserIcon, GithubIcon } from '@patternfly/react-icons';

const summary: string =
  'This task syncs (deploys) an Argo CD application and waits for it to be healthy.';
const detail: string =
  'To do so, it requires the address of the Argo CD server and some form of authentication either a username/password or an authentication token.';

const Details: React.FC = () => {
  return (
    <Grid
      style={{
        // width: '115%',
        margin: '-2em',
        paddingBottom: '2em'
      }}
    >
      <GridItem>
        <Card>
          <Grid>
            <GridItem span={1} />
            <GridItem span={1}>
              <div style={{ marginTop: '4.7em' }}>
                <BuildIcon size="xl" />
              </div>
            </GridItem>
            <GridItem span={9}>
              <CardHeader style={{ paddingTop: '2em', marginLeft: '-4em' }}>
                <TextContent>
                  <Grid span={2} style={{ marginTop: '1.5em' }} hasGutter>
                    <GridItem span={1}>
                      <Text component={TextVariants.h2} style={{ fontSize: '2em' }}>
                        argocd
                      </Text>
                    </GridItem>
                    <GridItem span={1} style={{ marginTop: '0.3em' }}>
                      <UserIcon size="lg" />
                    </GridItem>
                  </Grid>
                  <GithubIcon size="md" style={{ marginRight: '0.5em' }} />
                  <a href="github.com/tektoncd/catalog" target="_">
                    Open argocd in Github
                  </a>
                  <Grid span={8}>
                    <GridItem style={{ textAlign: 'justify' }}>
                      {summary}
                      <br />
                      <br />
                      {detail}
                    </GridItem>
                    <GridItem>
                      <Badge
                        style={{
                          paddingRight: '1em',
                          marginBottom: '1em',
                          marginRight: '1em'
                        }}
                        key="cli"
                        className="badge"
                      >
                        cli
                      </Badge>
                      <Badge
                        style={{
                          paddingRight: '1em',
                          marginBottom: '1em',
                          marginRight: '1em'
                        }}
                        key="tekton"
                        className="badge"
                      >
                        tekton
                      </Badge>
                    </GridItem>
                  </Grid>
                </TextContent>
              </CardHeader>
            </GridItem>
            <GridItem span={1} />
          </Grid>
        </Card>
      </GridItem>
    </Grid>
  );
};
export default Details;
*/

import React from 'react';
import {
  Card,
  CardHeader,
  Grid,
  GridItem,
  Text,
  TextContent,
  TextVariants,
  Badge,
  CardHeaderMain,
  CardActions,
  CardFooter,
  Button,
  Dropdown,
  DropdownToggle
} from '@patternfly/react-core';
import { BuildIcon, UserIcon, GithubIcon } from '@patternfly/react-icons';
import Rating from '../Rating/Rating';

const summary: string =
  'This task syncs (deploys) an Argo CD application and waits for it to be healthy.';
const detail: string =
  'To do so, it requires the address of the Argo CD server and some form of authentication either a username/password or an authentication token.';
const dropdownItems: any = [];
const Details: React.FC = () => {
  return (
    <Grid
      style={{
        margin: '-2em'
      }}
    >
      <GridItem>
        <Card>
          <CardHeader style={{ marginTop: '1.5em' }}>
            <CardHeaderMain>
              <Grid style={{ marginTop: '2.5em' }}>
                <GridItem span={1} />
                <GridItem span={1}>
                  <BuildIcon size="xl" color="#484848" />
                </GridItem>
                <GridItem span={1}>
                  <Text component={TextVariants.h2} style={{ fontSize: '2em' }}>
                    argocd
                  </Text>
                </GridItem>
                <GridItem span={9} style={{ marginTop: '0.3em' }}>
                  <UserIcon size="lg" />
                </GridItem>
                <GridItem span={2} />
                <GridItem span={10}>
                  <TextContent>
                    <GithubIcon size="md" style={{ marginRight: '0.5em' }} />
                    <a href="github.com/tektoncd/catalog" target="_">
                      Open argocd in Github
                    </a>
                    <Grid span={8}>
                      <GridItem style={{ textAlign: 'justify' }}>
                        {summary}
                        <br />
                        <br />
                        {detail}
                      </GridItem>
                    </Grid>
                  </TextContent>
                </GridItem>
              </Grid>
            </CardHeaderMain>
            <CardActions>
              <Grid>
                <GridItem>
                  <Rating />
                </GridItem>
                <GridItem>
                  <Button variant="primary" className="button" style={{ width: '8.5em' }}>
                    Install
                  </Button>
                </GridItem>
                <GridItem>
                  <Dropdown
                    style={{ marginLeft: '-1em', marginTop: '2em' }}
                    toggle={
                      <DropdownToggle style={{ width: '8.5em' }}>0.1 (latest)</DropdownToggle>
                    }
                    dropdownItems={dropdownItems}
                  />
                </GridItem>
              </Grid>
            </CardActions>
          </CardHeader>
          <CardFooter>
            <Grid span={12}>
              <GridItem style={{ paddingLeft: '16.5em' }}>
                <Badge
                  style={{
                    paddingRight: '1em',
                    marginBottom: '1em',
                    marginRight: '1em'
                  }}
                  key="cli"
                  className="badge"
                >
                  cli
                </Badge>
                {/* </GridItem>
              <GridItem span={1}> */}
                <Badge
                  style={{
                    paddingRight: '1em',
                    marginBottom: '1em',
                    marginRight: '1em'
                  }}
                  key="tekton"
                  className="badge"
                >
                  tekton
                </Badge>
              </GridItem>
            </Grid>
            {/* <div style={{ height: '2.5em', marginLeft: '16.5em' }}>
              
            </div> */}
          </CardFooter>
        </Card>
      </GridItem>
    </Grid>
  );
};
export default Details;
