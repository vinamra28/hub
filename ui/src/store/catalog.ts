import { types, Instance } from 'mobx-state-tree';

export const CatalogType = types
  .model({
    id: types.integer,
    name: types.string,
    selected: false
  })
  .actions((self) => ({
    toggle() {
      self.selected = !self.selected;
    }
  }));

export type ICatalogType = Instance<typeof CatalogType>;
export type ICatalogStore = Instance<typeof CatalogStore>;

export const CatalogStore = types
  .model('CatalogStore', {
    list: types.array(CatalogType)
  })

  .views((self) => ({
    get count(): number {
      return self.list.length;
    },
    get catalogs(): Array<string> {
      const { list } = self;
      return list
        .filter((type: ICatalogType) => type.selected)
        .map((type: ICatalogType) => type.name);
    }
  }))

  .actions((self) => ({
    add(item: ICatalogType) {
      self.list.push(item);
    },
    clear() {
      self.list.map((item: ICatalogType) => (item.selected = false));
    }
  }));
