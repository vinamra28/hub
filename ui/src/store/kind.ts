import { types, Instance } from 'mobx-state-tree';

export const Kind = types
  .model('Kind', {
    id: types.integer,
    name: types.string,
    selected: false
  })
  .actions((self) => ({
    toggle() {
      self.selected = !self.selected;
    }
  }));

export type IKind = Instance<typeof Kind>;
export type IKindStore = Instance<typeof KindStore>;

export const KindStore = types
  .model('KindStore', {
    list: types.array(Kind)
  })

  .views((self) => ({
    get count(): number {
      return self.list.length;
    },
    get kinds(): Array<string> {
      const { list } = self;
      return list.filter((type: IKind) => type.selected).map((type: IKind) => type.name);
    }
  }))

  .actions((self) => ({
    add(item: IKind) {
      self.list.push(item);
    },
    clear() {
      self.list.map((item: IKind) => (item.selected = false));
    }
  }));
