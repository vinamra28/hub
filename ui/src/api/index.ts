import axios from 'axios';
import { API_URL } from '../config/constants';
import { ICategory } from '../store/category';
import { IResource } from '../store/resource';

export interface Api {
  categories(): Promise<ICategory>;
  resources(): Promise<IResource>;
}

export class Hub implements Api {
  async categories() {
    try {
      return axios.get(`${API_URL}/categories`).then((response) => response.data);
    } catch (err) {
      return err.response;
    }
  }

  async resources() {
    try {
      return axios.get(`${API_URL}/resources`).then((response) => response.data);
    } catch (err) {
      return err.response;
    }
  }

  async resourceVersion(resourceId: string) {
    try {
      return axios
        .get(`${API_URL}/resource/${resourceId}/versions`)
        .then((response) => response.data);
    } catch (err) {
      return err.response;
    }
  }
}
