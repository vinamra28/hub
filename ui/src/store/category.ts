import { types, getEnv, flow, Instance } from 'mobx-state-tree';
import { Api } from '../api';

export const Tag = types.model('Tags', {
  id: types.integer,
  name: types.string
});

export const Category = types
  .model('Category', {
    id: types.number,
    name: types.string,
    tags: types.array(Tag),
    selected: false
  })
  .actions((self) => ({
    toggle() {
      self.selected = !self.selected;
    }
  }));

export type ICategory = Instance<typeof Category>;
export type ITag = Instance<typeof Tag>;

export const CategoryStore = types
  .model('CategoryStore', {
    items: types.array(Category),
    isLoading: true,
    err: ''
  })

  .views((self) => ({
    get api(): Api {
      return getEnv(self).api;
    },

    get count() {
      return self.items.length;
    },

    get tags() {
      return self.items
        .filter((c) => c.selected)
        .reduce((acc: string[], c: ICategory) => [...acc, ...c.tags.map((t) => t.name)], []);
    }
  }))

  .actions((self) => ({
    add(item: ICategory) {
      self.items.push(item);
    },

    setLoading(l: boolean) {
      self.isLoading = l;
    },

    clear() {
      self.items.map((c: ICategory) => (c.selected = false));
    }
  }))

  .actions((self) => ({
    load: flow(function* () {
      try {
        self.setLoading(true);
        const { api } = self;
        const json = yield api.categories();
        json.data.forEach((item: ICategory) => self.add(item));
      } catch (err) {
        self.err = err.toString();
      }
      self.setLoading(false);
    })
  }))

  .actions((self) => ({
    afterCreate() {
      self.load();
    }
  }));

export type ICategoryStore = Instance<typeof CategoryStore>;
