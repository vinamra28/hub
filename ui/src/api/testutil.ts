import { Api } from './';
import * as fs from 'fs';
import { ICategory } from '../store/category';
import { IResource } from '../store/resources';

export class FakeHub implements Api {
  dataDir: string;

  constructor(dataDir: string) {
    this.dataDir = dataDir;
  }

  async resources() {
    const data = `${this.dataDir}/resources.json`;

    const ret = () => JSON.parse(fs.readFileSync(data).toString());
    return new Promise<IResource>((resolve) => {
      setTimeout(() => resolve(ret()), 1000);
    });
  }

  async categories() {
    const data = `${this.dataDir}/categories.json`;

    const ret = () => JSON.parse(fs.readFileSync(data).toString());
    return new Promise<ICategory>((resolve) => {
      setTimeout(() => resolve(ret()), 1000);
    });
  }
}
