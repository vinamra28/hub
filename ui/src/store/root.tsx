import React, { ReactChild, ReactChildren } from 'react';
import { types, getEnv, Instance } from 'mobx-state-tree';
import { CategoryStore } from './category';
import { ResourceStore } from './resource';
import { Hub, Api } from '../api';

export const Root = types.model('Root', {}).views((self) => ({
  get api(): Api {
    return getEnv(self).api;
  },
  get categories() {
    return getEnv(self).categories;
  },
  get resources() {
    return getEnv(self).resources;
  }
}));

type IRoot = Instance<typeof Root>;

const initRootStore = (api: Api) => {
  const categories = CategoryStore.create({}, { api });
  const resources = ResourceStore.create({}, { api, categories });
  return Root.create({}, { api, categories, resources });
};

interface Props {
  children: ReactChild | ReactChildren;
}

let RootContext: React.Context<IRoot>;
export const useMst = () => React.useContext(RootContext);

export const createProviderAndStore = (api?: Api) => {
  const root = initRootStore(api || new Hub());
  RootContext = React.createContext(root);

  const Provider = ({ children }: Props) => (
    <RootContext.Provider value={root}> {children} </RootContext.Provider>
  );
  return { Provider, root };
};

export const createProvider = (api?: Api) => {
  const { Provider } = createProviderAndStore(api);
  return Provider;
};
