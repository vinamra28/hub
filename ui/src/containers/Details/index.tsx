import React from 'react';
import { useMst } from '../../store/root';
import { useObserver } from 'mobx-react';
import { Spinner } from '@patternfly/react-core';
import { useParams } from 'react-router-dom';

const Details: React.FC = () => {
  const { resources } = useMst();
  const { name } = useParams();

  return useObserver(() =>
    resources.resources.size === 0 ? (
      <Spinner className="hub-spinner" />
    ) : (
      <React.Fragment>
        <span>Add Resources detail here {resources.resources.get(name).name}</span>
      </React.Fragment>
    )
  );
};
export default Details;
