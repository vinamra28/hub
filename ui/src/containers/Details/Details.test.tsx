import React from 'react';
import Details from '.';
import { FakeHub } from '../../api/testutil';
import { createProviderAndStore } from '../../store/root';
import { mount } from 'enzyme';
import { when } from 'mobx';

const TESTDATA_DIR = `src/store/testdata`;
const api = new FakeHub(TESTDATA_DIR);
const { Provider, root } = createProviderAndStore(api);

jest.mock('react-router-dom', () => {
  return {
    useParams: () => {
      return {
        name: 'ansible-runner'
      };
    }
  };
});

it('should render the details component', (done) => {
  const component = mount(
    <Provider>
      <Details />
    </Provider>
  );

  const { resources } = root;
  when(
    () => {
      return !resources.isLoading;
    },
    () => {
      setTimeout(() => {
        const resource = resources.filteredResources;
        expect(resource.length).toBe(6);

        component.update();

        const r = component.find(Details);
        expect(r.length).toEqual(1);

        expect(component.debug()).toMatchSnapshot();

        done();
      }, 1000);
    }
  );
});
