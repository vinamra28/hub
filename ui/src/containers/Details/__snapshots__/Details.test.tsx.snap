// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the details component 1`] = `
"<Provider>
   
  <Details>
    <span>
      Add Resources detail here 
      ansible-runner
    </span>
  </Details>
   
</Provider>"
`;
